### PR TITLE
ENG-6164 view registration tests 

### PR DESCRIPTION
**Pull request by kishisaka-nid**
view registration tests and small refactor to send in logger and data source to allow mocking.
test for all widget types except the spinner, still working on this one.

### CODE REVIEW

File: NeuroID/src/androidLib/java/com/neuroid/tracker/events/NIDIdentifyAllViews.kt [hunk 1]
```diff
@@ -29,27 +29,31 @@ import org.json.JSONObject
 fun identifyAllViews(
     viewParent: ViewGroup,
     guid: String,
+    logger: NIDLogWrapper,
+    storeManager: NIDDataStoreManager,
```
Comment (kevinsites) [inline NeuroID/src/androidLib/java/com/neuroid/tracker/events/NIDIdentifyAllViews.kt:33]: I don't see this being used anywhere?

Comment (kishisaka-nid) [reply NeuroID/src/androidLib/java/com/neuroid/tracker/events/NIDIdentifyAllViews.kt:33]: These are used. logger is used in place of NIDLogger and storeManager in place of getDataStoreInstance().

Comment (kishisaka-nid) [reply NeuroID/src/androidLib/java/com/neuroid/tracker/events/NIDIdentifyAllViews.kt:33]: Have to keep the storeManager mock for the tests to complete.

**Comment by kishisaka-nid** [comment]
merge to master is complete and seeing the react image view buttons registrations in logs:  `"et":"ReactButton::ReactImageView::ReactViewGroup"`

**Comment by kishisaka-nid** [comment]
All unit tests passed on first try. AMAZING!

**Review by kevinsites — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Review by cometman — APPROVED** [review]

**Pull request merged**